src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt

```diff
@@ -0,0 +1,67 @@
+package com.sipios.springsearch.strategies
+
+import com.sipios.springsearch.SearchOperation
+import java.util.Date
+import javax.persistence.criteria.CriteriaBuilder
+import javax.persistence.criteria.Path
+import javax.persistence.criteria.Predicate
+import kotlin.reflect.KClass
+
+interface ParsingStrategy {
+    /**
+     * Method to parse the value specified to the corresponding strategy
+     *
+     * @param value Value used for the search
+     * @param fieldClass Kotlin class of the referred field
+     * @return Returns by default the value without any parsing
+     */
+    fun parse(value: String?, fieldClass: KClass<out Any>): Any? {
+        return value
+    }
+
+    /**
+     * Method to build the predicate
+     *
+     * @param builder Criteria object to build on
+     * @param path Current path for predicate
+     * @param fieldName Name of the field to be searched
+     * @param ops Search operation to use
+     * @param value Value used for the search
+     * @return Returns a Predicate instance or null if the operation was not found
+     */
+    fun buildPredicate(
+        builder: CriteriaBuilder,
+        path: Path<*>,
+        fieldName: String,
+        ops: SearchOperation?,
+        value: Any?
+    ): Predicate? {
+        return when (ops) {
+            SearchOperation.EQUALS -> builder.equal(path.get<Any>(fieldName), value)
+            SearchOperation.NOT_EQUALS -> builder.notEqual(path.get<Any>(fieldName), value)
+            SearchOperation.STARTS_WITH -> builder.like(path.get(fieldName), "$value%")
+            SearchOperation.ENDS_WITH -> builder.like(path.get(fieldName), "%$value")
+            SearchOperation.CONTAINS -> builder.like((path.get<String>(fieldName).`as`(String::class.java)), "%$value%")
```

Comment:
(In addition to my other comment) The `.lower` got removed in this specific line, and in the SearchOperation `DOESNT_CONTAIN`.

Reply:
See response above :)